dowhy/gcm/influence.py

```diff
@@ -0,0 +1,339 @@
+                              Callable[[np.ndarray, np.ndarray], Union[np.ndarray, float]],
+                              num_samples_conditional: int,
+                              max_num_runs: int,
+                              tolerance: float) -> float:
+    distribution_samples = shape_into_2d(distribution_samples)
+
+    aggregated_conditional_difference_result = 0
+    average_difference_result = 0
+    converged_run = 0
+    for run, sample in enumerate(distribution_samples):
+        tmp_samples = repmat(sample, num_samples_conditional, 1)
+
+        rnd_permutation = np.random.choice(distribution_samples.shape[0],
+                                           num_samples_conditional,
+                                           replace=False)
+
+        removed_arrow_sampling = shape_into_2d(distribution_samples[:, target_columns][rnd_permutation])
+        conditional_distribution_samples = draw_samples_func(tmp_samples)
+        tmp_samples[:, target_columns] = removed_arrow_sampling
+        cond_dist_removed_arr_samples = draw_samples_func(tmp_samples)
+
+        old_average_difference_result = average_difference_result
+
+        aggregated_conditional_difference_result += difference_estimation_func(conditional_distribution_samples,
+                                                                               cond_dist_removed_arr_samples)
+
+        average_difference_result = aggregated_conditional_difference_result / (run + 1)
+
+        if run >= max_num_runs:
+            break
+        elif run > 0:
+            if old_average_difference_result == 0:
+                old_average_difference_result = EPS
+
+            if abs(1 - average_difference_result / old_average_difference_result) < tolerance:
+                converged_run += 1
+                if converged_run >= 3:
+                    break
+            else:
+                converged_run = 0
+
+    return average_difference_result
+
+
+def intrinsic_causal_influence(causal_model: StructuralCausalModel,
+                               target_node: Any,
+                               prediction_model: Union[PredictionModel, ClassificationModel, str] = 'exact',
+                               attribution_func: Optional[Callable[[np.ndarray, np.ndarray], float]] = None,
+                               num_training_samples: int = 100000,
+                               num_noise_samples: int = 7500,
+                               num_evaluation_samples: int = 1000,
+                               max_batch_size: int = 100,
+                               approx_select_model: Callable[[np.ndarray, np.ndarray],
+                                                             Union[PredictionModel, ClassificationModel]] = None,
+                               shapley_config: Optional[ShapleyConfig] = None) -> Dict[Any, float]:
+    """Computes the causal contribution of each noise term in the functional causal model to the statistical property
```

Comment:
is it noise term of all nodes or just the noise term of the parents/ancestors of target node?

Reply:
Only the noise terms of ancestors of the target (including the noise if the target itself). All other noise terms, wouldn't have a causal influence on the target. I clarify this in the revision.